app/assets/javascripts/controllers/live_metrics/util/metrics-utils-factory.js

```diff
@@ -102,17 +102,43 @@ angular.module('miq.util').factory('metricsUtilsFactory', function() {
         data[0].value.toFixed(2) + '</div>';
     };
 
-    var metricPrefix = function(maxValue) {
-      var exp = parseInt(Math.log10(maxValue) / 3, 10) * 3;
-      var metricPrefixes = {
-        3: {prefix: 'k', divisor: Math.pow(10, 3)},
-        6: {prefix: 'M', divisor: Math.pow(10, 6)},
-        9: {prefix: 'G', divisor: Math.pow(10, 9)},
-        12: {prefix: 'T', divisor: Math.pow(10, 12)},
-        15: {prefix: 'P', divisor: Math.pow(10, 15)}
+    var metricPrefix = function(maxValue, units) {
+      var metricPrefixes;
+      var baseUnit;
+      var baseUnitMultiplier;
+      var baseUnitMaxValue;
+      var exp;
+
+      // get base unit for special case units
+      switch (units) {
+        case 'milliseconde':
+        case 'ms':
+          baseUnitMultiplier = Math.pow(10, -3);
+          baseUnit = 's';
+          break;
+        case 'ns':
+          baseUnitMultiplier = Math.pow(10, -9);
+          baseUnit = 's';
+          break;
+        default:
+          baseUnitMultiplier = 1;
+          baseUnit = units;
+      }
+
+      // adjuat to base units and calc exponent
+      baseUnitMaxValue = maxValue * baseUnitMultiplier;
+      exp = parseInt(Math.log10(baseUnitMaxValue) / 3, 10) * 3;
```

Comment:
`parseInt` to convert from float to int? Sounds like `~~` might do the trick better..